domaincommon/src/state.rs

```diff
@@ -246,7 +306,8 @@ impl Aggregate for Match {
                 }))
             }
             GameEvent::MatchTurnCompleted { new_turn } => {
-                Ok(MatchState::advance_match_turn(state, *new_turn))
+                let state = MatchState::reset_mech_actions(state);
+                Ok(MatchState::advance_match_turn(&state, *new_turn))
```

Comment:
resetting the mech action points is logically part of the work of advancing the match turn... it might be more readable if we move the call to reset_mech_actions inside the `advance_match_turn` function.